openfl/component/aggregator/aggregator.py

```diff
@@ -331,11 +336,64 @@ def get_tasks(self, collaborator_name):
         )
         sleep_time = 0
 
-        if hasattr(self.straggler_handling_policy, 'round_start_time'):
-            self.straggler_handling_policy.round_start_time = time.time()
+        # For CutoffTimeBasedStragglerHandling start straggler handling policy
+        # and pass callback function
+        if hasattr(self.straggler_handling_policy, "round_start_time"):
+            self.straggler_handling_policy.start_policy(
+                self._straggler_cutoff_time_elapsed, collaborator_name
+            )
+        else:
+            self.straggler_handling_policy.start_policy(
+                None, collaborator_name
+            )
 
         return tasks, self.round_number, sleep_time, time_to_quit
 
+    def _straggler_cutoff_time_elapsed(self) -> None:
+        """
+        This method is called by the straggler handling policy when cutoff timer is elapsed.
+        It applies straggler handling policy and ends the round early.
+
+        Returns:
+            None
+        """
+        # Stop timer from restarting until next round
+        self.straggler_handling_policy.is_timer_expired = True
```

Comment:
It would be cleaner if the policy object itself sets `is_timer_expired` to `True`. It doesn't seem safe to leave this responsibility with the callback function, as the implementor may not be aware of it.

Reply:
Accepted, the policy object now manages the flag.
NOTE: is_timer_expired is renamed to __is_policy_applied_for_round, and the flag is managed within the class.